### PR TITLE
[WIP] Add openshift e2e test

### PR DESCRIPTION
**Pull request by kargakis**
You can test deploying an OpenShift cluster with this:
```shell
export SKIP_TEST=true
export LOCATION=eastus
export ORCHESTRATOR_RELEASE=3.9
export CLUSTER_DEFINITION=examples/openshift.json
export CREATE_VNET=false
export RETAIN_SSH=true
export CLEANUP_ON_EXIT=false
export SUBSCRIPTION_ID=<>
export CLIENT_ID=<>
export CLIENT_SECRET=<>
export TENANT_ID=<>
make test-openshift
```
@jim-minter @pweil- 

TODO:
- [ ] add ginkgo tests for testing things like running router/registry/etc

### CODE REVIEW

**Comment by pweil-** [comment]
Nice, thanks for starting in on this

**Comment by kargakis** [comment]
I forgot to add a `export RETAIN_SSH=true` in the initial post that should leave the ssh keys around after the test is done so you can ssh into the master. Updating the htpasswd credentials to demo/demo, too, to make it easier to play around with this for now.

**Comment by kargakis** [comment]
We can also start testing with custom vnets I think.

File: .circleci/config.yml [hunk 1]
```diff
@@ -336,6 +336,38 @@ jobs:
...
+        environment:
+          GOPATH: /go
+    steps:
+      - checkout
+      - run: |
+          echo 'export TIMEOUT=20m' >> $BASH_ENV
+          echo 'export SKIP_TEST=true' >> $BASH_ENV
+          echo 'export LOCATION=eastus' >> $BASH_ENV
+          echo 'export ORCHESTRATOR_RELEASE=3.9' >> $BASH_ENV
+          echo 'export CLUSTER_DEFINITION=examples/openshift.json' >> $BASH_ENV
```
Comment (kargakis) [inline .circleci/config.yml]: @kwoodson this will make you happy; I am reusing our example for the e2e test.

**Comment by kargakis** [comment]
Depends on https://github.com/kargakis/acs-engine/pull/108

**Comment by kargakis** [comment]
Closing in favor of https://github.com/Azure/acs-engine/pull/2675

**Pull request closed without merging**